### PR TITLE
Adds initial FDIO support

### PR DESCRIPTION
**Pull request by trozet**

### CODE REVIEW

File: puppet/manifests/overcloud_controller.pp [hunk 1]
```diff
@@ -491,6 +491,36 @@
...
+"netconf-node-topology:max-connection-attempts":10,
+"netconf-node-topology:between-attempts-timeout-millis":10000,
+"netconf-node-topology:schema-cache-directory":"hcmount"}
+]
+}
+'
+          $fdio_data = inline_template($fdio_data_template)
+          $fdio_url = "http://${opendaylight_controller_ip}:${opendaylight_port}/restconf/config/network-topology:network-topology/network-topology:topology/topology-netconf"
+          exec { 'VPP Mount into ODL':
+            command   => "curl -o /dev/null --fail --silent -u ${odl_username}:${odl_password} ${fdio_url} -i -H 'Content-Type: application/json' --data \'${fdio_data}\' -X PUT",
```
Comment (michaeltchapman) [inline puppet/manifests/overcloud_controller.pp:519]: Is this safe to run multiple times? If there's a query that can determine if it's already been run, I think it's worth adding as an unless. Ditto for the other two files.

File: environments/opendaylight_fdio.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#  OpenDaylightUsername: admin
+#  OpenDaylightPassword: admin
+
+parameters:
+    # increase this if you need more ODL nodes
+    # OpenDaylightCount: 1
+    NeutronEnableForceMetadata: true
+    NeutronL3HA: false
+    OpenDaylightEnableHA: false
+    OpenDaylightFeatures: "odl-netconf-connector-all,old-vbd,odl-groupbasedpolicy-neutronmapper,odl-restconf"
```
Comment (wojd) [inline environments/opendaylight_fdio.yaml]: "old-vbd" should be "odl-vbd"

And replace: odl-groupbasedpolicy-neutronmapper
with
odl-groupbasedpolicy-neutron-vpp-mapper

File: puppet/manifests/overcloud_compute.pp [hunk 3]
```diff
@@ -259,6 +259,36 @@
...
         path => '/usr/local/bin:/usr/bin:/sbin:/bin:/usr/local/sbin:/usr/sbin',
       }
+    } elsif hiera('fdio', false) {
+      $odl_username  = hiera('opendaylight_username')
+      $odl_password  = hiera('opendaylight_password')
+      $ctrlplane_interface = hiera('nic1')
+      if ! $ctrlplane_interface { fail("Cannot map logical interface NIC1 to physical interface") }
+      $vpp_ip = inline_template("<%= scope.lookupvar('::ipaddress_${ctrlplane_interface}') -%>")
+      $fdio_data_template='{"node" : [
+{"node-id":"<%= @hostname %>",
```
Comment (wojd) [inline puppet/manifests/overcloud_compute.pp]: What does the @hostname resolve to?

File: puppet/manifests/overcloud_controller.pp (hunk 1 above)
Comment (trozet) [inline puppet/manifests/overcloud_controller.pp:519]: @michaeltchapman yeah we need an unless here.  In the unless we should query restconf on ODL and verify the VPP node was already mounted.

File: puppet/manifests/overcloud_compute.pp (hunk 3 above)
Comment (trozet) [inline puppet/manifests/overcloud_compute.pp]: @wojd hostname resolves to shorthand hostname of the server.  I need to look into this and make it fqdn

File: puppet/manifests/overcloud_controller.pp (hunk 1 above)
Comment (trozet) [inline puppet/manifests/overcloud_controller.pp:519]: @michaeltchapman given that this only runs one time, we can ignore the unless for now.  I will go back and add the unless once I have a test environment where I can verify hte correct command.

**Pull request merged**